hr_holidays_natural_period/models/resource_calendar.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2020 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import logging
+from datetime import datetime, time
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import models
+
+from odoo.addons.resource.models.resource import Intervals
+
+_logger = logging.getLogger(__name__)
+
+
+class ResourceCalendar(models.Model):
+    _inherit = "resource.calendar"
+
+    def _attendance_intervals_batch(
+        self, start_dt, end_dt, resources=None, domain=None, tz=None
+    ):
+        intervals = super()._attendance_intervals_batch(
+            start_dt=start_dt, end_dt=end_dt, resources=resources, domain=domain, tz=tz
+        )
+        if self.env.context.get("natural_period"):
+            for resource in resources:
+                if resource:
+                    interval_resource = intervals[resource.id]
+                    if len(interval_resource._items) > 0:
+                        dt_item = start_dt.date()
+                        new_intervals = []
+                        while dt_item != end_dt.date():
```

Comment:
A safer option is to compare this way:
```suggestion
                        while dt_item < end_dt.date():
```